driver/makedepend.ml

```diff
@@ -326,7 +327,7 @@ let read_parse_and_extract parse_function extract_function def ast_kind
     end
   with x -> begin
     print_exception x;
-    if not !allow_approximation then begin
+    if !strict || not !allow_approximation then begin
```

Comment:
Do I understand correctly that in `-strict` mode the tool will exit with a non-zero code also if the file contains a syntax error or some other is raised when processing it (not just if the file does not exist)? If yes, the documentation should be adapted accordingly.